python/rikai/spark/sql/codegen/mlflow_logger.py

```diff
@@ -131,6 +132,30 @@ def log_model(
                     "will not be populated to MLflow"
                 )
         mlflow.set_tags(tags)
+        if registered_model_name is not None:
+            # if we're creating a model registry entry,
+            # we also want to set the tags on the model version
+            # and model to enable search
+            c = MlflowClient()
+            # mlflow log_model does not return the version (wtf)
+            all_versions = c.get_latest_versions(
+                registered_model_name, stages=["production", "staging", "None"]
+            )
+            current_version = None
+            run_id = mlflow.active_run().info.run_id
+            for v in all_versions:
+                if v.run_id == run_id:
+                    current_version = v
+                    break
+            if current_version is None:
+                raise ValueError(
+                    "No model version found matching runid: {}".format(run_id)
+                )
+            for key, value in tags.items():
+                c.set_registered_model_tag(registered_model_name, key, value)
```

Comment:
if the previous version logged some tags that's removed in this version, is there no way to detect/remove?

Reply:
It's possible, but we need to define the `previous version` first. Is it the previous version of current model name? It's possible the previous version is generated by other unrelated runs.
And then, maybe we should define a rikai related tags scope like what tags are managed by rikai to prevent other tags accidentally deleted by rikai

Reply:
Please create a GH issue to deal with the scoping (e.g., use a prefix, turn the rikai tags into a single dict tag). Please describe the problem fully and link to this so later people know what it's about